sig-cli/contributing.md

```diff
@@ -0,0 +1,311 @@
+contains the GitHub handles for the sig-cli leads, slack channels,
+email discussion group, and bi-weekly meeting time.
+
+## Message to new contributors
+
+Welcome to the Kubernetes sig-cli contributing guide.  We are excited
+about the prospect of you joining our community.  Please understand
+that all contributions to Kubernetes require time and commitment from
+the project maintainers to review the ux, software design, and code.  We
+recommend that you reach out to one of the sig-leads
+and ask the best way to get started.  It is likely that there a couple
+issues perfect for a new contributor to get started on.
+
+Sig-leads can be reached in the sig-cli slack channel or at the sig-cli
+bi-weekly meeting.  See the [sig-cli community page](https://github.com/kubernetes/community/tree/master/sig-cli)
+for a link to the slack channel, list of sig-leads, and the community meeting time and location.
+
+Example message for asking how to become a contributor:
+
+    Hello, my name is <your name>.  I would like to get involved in
+    contributing to the Kubernetes project.  What is the best way to
+    get started?
+
+Please understand that mentoring and on boarding new contributors takes
+a lot time and energy from maintainers who have many other
+responsibilities.
+
+## Before you begin
+
+- Visit the [sig-cli community page](https://github.com/kubernetes/community/tree/master/sig-cli)
+  - Join the `kubernetes-sig-cli@googlegroups.com` so you get email updates
+  - Join the Kubernetes slack channel `sig-cli`
+- Introduce yourself to the community by sending an email to kubernetes-sig-cli@googlegroups.com and let us know you want to be a contributor
+
+## Finding an existing issue to work on
+
+The preferred way to own a piece of work is to directly reach out
+to one of the sig-cli leads.  This will ensure that the issue is
+high-priority enough that it will receive PR review bandwidth from
+the sig-cli community.
+
+1. In the Kubernetes sig-cli slack channel, mention @pwittrock, @adohe, or  @fabianofranz and ask if there are any issues you could pick up
+2. Attend the sig-cli [bi-weekly meeting](https://github.com/kubernetes/community/tree/master/sig-cli).
+  - Introduce yourself at the beginning of the meeting
+
+## Life of a sig-cli bug overview
+
+    Picking up a bug and implementing a fix without discussing the scope
+    and requirements for the fix can result in wasted time
+    and effort due decisions such as the propper way to the fix the
+    underlying issue and scope of the changes.
+
+    Small 10-20 (non-test) line fixes for issues with clear solutions do not need
+    extensive discussion and can be implemented directly.
+
+    Issues requiring complex fixes or > 50 lines of (non-test) code should
+    be added to the sig-cli meeting agenda and briefly discussed
+    before an implementation PR is sent for review.  This will help
+    to identify any issues with the solution before the code review.
```

Comment:
I wanted to make it visually distinct from the next section.  Will dedent if it is confusing.